### ISSUE TITLE
Select whole line from mouse position on triple-click 🔥

### ISSUE THREAD

**Issue by silverqx** [microsoft/terminal#9767]
I really miss this feature from the _Konsole_ terminal, it has this configuration in settings 🚀:

![triple_click](https://user-images.githubusercontent.com/86900/114264810-4e9ad180-99ed-11eb-9ee8-c9be085be698.jpg)

It works this way, if I do triple-click on some word, it selects the word under the cursor and the whole line to the end.

Examples:

- `|` indicates where I do triple-click
- `>` start of the selection after triple-click
- `<` end of the selection after triple-click

```
Setting up >pyth|on3-distupgrade (1:20.04.31) ...                             <
>Se|tting up python3-distupgrade (1:20.04.31) ...                             <
Setting up >python3-distupgra|de (1:20.04.31) ...                             <
```

It can be a little enhanced like this:

```
Setting up >pyth|on3-distupgrade (1:20.04.31) ...<
>Se|tting up python3-distupgrade (1:20.04.31) ...<
Setting up >python3-distupgra|de (1:20.04.31) ...<
```

So do the trim of whitespaces at the end of the line.

Would be nice to do it configurable, so the user could decide if he wants to select `Whole line` or `From mouse position to the end of line`.

I'm using this feature heavily when I work in the _Konsole_ terminal.

**Comment by j4james** [comment]
I think this could probably be covered by issue #6511. It's essentially just another selection type that users would be able to choose for a given click action.

**Comment by DHowett** [comment]
Yep! Thanks @j4james. This is a solid /duplicate of #6511, but for us to correctly identify the righthandmost text we'll need to fix #2089.

Thanks!
